onnx/version_converter/adapters/softmax_12_13.h

```diff
@@ -48,6 +48,10 @@ class Softmax_12_13 final : public Adapter {
 
         // Set shape input of Reshape
         const std::vector<Dimension>& target_shape = flatten->inputs()[0]->sizes();
+
+        ONNX_ASSERTM(target_shape.size() != 0, "Version conversion for Softmax failed because "
+          "input shape is unknown.");
+
```

Comment:
Not sure whether the version conversion here for Softmax can be done without knowing the input shape due to dynamic situation... At least version converter should catch this failure and not produce an invalid model. (has a Reshape node with an empty Shape)

Reply:
True. This problem also appears elsewhere, since the version-converter seems to assume that the shapes are known in many places. How does the underlying IR represent the cases where the rank is unknown or a dimension is unknown?

Reply:
IIUC, it will create a node with empty sizes() (empty vector of Dimension).